conan/tools/meson/meson.py

```diff
@@ -15,10 +16,18 @@ def configure(self, reconfigure=False):
         generators_folder = self._conanfile.generators_folder
         cross = os.path.join(generators_folder, MesonToolchain.cross_filename)
         native = os.path.join(generators_folder, MesonToolchain.native_filename)
+        deps_flags = os.path.join(generators_folder, MesonDeps.filename)  # extra machine files layer
+        has_deps_flags = os.path.exists(deps_flags)
+
         if os.path.exists(cross):
             cmd += ' --cross-file "{}"'.format(cross)
+            if has_deps_flags:
+                cmd += ' --cross-file "{}"'.format(deps_flags)
         else:
             cmd += ' --native-file "{}"'.format(native)
+            if has_deps_flags:
+                cmd += ' --native-file "{}"'.format(deps_flags)
```

Comment:
but it overrides, isn't it?

Reply:
No, it's not. They are being added as an extended part of the MesonToolchain:

```
    c_args = {{c_args}} + preprocessor_definitions + c_args
    c_link_args = {{c_link_args}} + c_link_args
    cpp_args = {{cpp_args}} + preprocessor_definitions + cpp_args
    cpp_link_args = {{cpp_link_args}} + cpp_link_args
```

So, the only part that is being overridden, it's the first definition of the constants:
```
    [constants]
    ........
    c_args = []
    c_link_args = []
    cpp_args = []
    cpp_link_args = []
```

Reply:
I am not sure this won't be breaking. The values from multiple machine files override each other, only last file wins. If MesonDeps adds some flags, and MesonToolchain too, they will remove the previously defined one. I don't think the idiom ``c_args = {{c_args}} + preprocessor_definitions + c_args`` will work. Am I misunderstanding something?
Most likely a test that covers this is necessary.

Reply:
Yes, this is not working so. AutotoolDeps, as MesonDeps, is getting altogether the flags from all the dependencies so I'm creating only one file, the `conan_meson_deps_flags.ini`. It's not created a file per dependency.

Reply:
And, the test is already there for sure. That test is not passing without those deps flags. I can try to debug something from the log or try to print anything else.

Reply:
It is not a problem of one file per dependency. It will be the ``conan_meson_deps_flags.ini``, being overriden by the ``conan_meston_toolchain.ini`` file values for those cppflags, etc.

Reply:
IMO this complicates local flow significantly. now, if one wants to replicate conan buid exactly, will have to type something like `--native-file conan_meson_deps_flags1.ini --native-file conan_meson_deps_fags2.ini ... --native-file conan_meson_native.ini`, and it has to be in the correct order (besides the fact it's a long command line). I would just write everything into `conan_meson_native.ini` to simplify the workflow and reduce a chance of human error.

Reply:
No, you'll only have a single file for all the deps: `--native-file conan_meson_deps_flags.ini`, and that's it. Have a look at the last test that I've added.